source/lac/trilinos_vector.cc

```diff
@@ -199,18 +199,11 @@ namespace TrilinosWrappers
 
     void
     Vector::reinit (const Epetra_Map &input_map,
-                    const bool        omit_zeroing_entries)
+                    const bool)
     {
       nonlocal_vector.reset();
 
-      if (vector->Map().SameAs(input_map)==false)
-        vector.reset (new Epetra_FEVector(input_map));
-      else if (omit_zeroing_entries == false)
-        {
-          const int ierr = vector->PutScalar(0.);
-          (void)ierr;
-          Assert (ierr == 0, ExcTrilinosError(ierr));
-        }
+      vector.reset (new Epetra_FEVector(input_map));
```

Comment:
That's right. In case there was a barrier, we would really need to add the change to GrowingVectorMemory. So I think that should be enough for now.